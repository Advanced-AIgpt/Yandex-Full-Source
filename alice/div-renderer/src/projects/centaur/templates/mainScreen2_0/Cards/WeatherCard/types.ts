import { IAbstractCardProps } from '../types';

export interface IWeatherCardProps extends IAbstractCardProps {
    type: 'weather';
    city?: string | null;
    temperature?: number | null;
    image?: string | null;
    comment?: string | null;
    bgImage?: string | null;
}
