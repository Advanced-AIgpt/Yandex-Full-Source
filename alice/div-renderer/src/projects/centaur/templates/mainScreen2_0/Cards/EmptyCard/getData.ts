import { IEmptyCardProps } from './types';
import { ICardDataAdapter } from '../types';

export const getEmptyCardData: ICardDataAdapter<IEmptyCardProps> = function getEmptyCardData(card, requestState) {
    if (card && card.VacantCardData) {
        return {
            type: 'empty',
            requestState,
        };
    }
    return null;
};
