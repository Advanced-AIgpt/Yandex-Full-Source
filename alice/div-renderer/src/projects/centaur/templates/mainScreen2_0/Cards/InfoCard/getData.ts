import { sample } from 'lodash';
import { IInfoCardProps } from './types';
import { DISCOVERY_CARDS } from '../../../mainScreen/discovery';
import { ICardDataAdapter } from '../types';
import { textAction } from '../../../../../../common/actions';

export const getInfoCardData: ICardDataAdapter<IInfoCardProps> = function getInfoCardData(card, requestState) {
    if (typeof card.InfoCardData !== 'undefined' && card.InfoCardData !== null) {
        const data = card.InfoCardData.Title === 'test' ? DISCOVERY_CARDS[0] : sample(DISCOVERY_CARDS) || DISCOVERY_CARDS[0];

        return {
            type: 'info',
            title: data.title,
            image_background: data.bg,
            description: data.description,
            requestState,
            actions: [
                {
                    log_id: 'discovery.mainscreen.execute',
                    url: textAction(data.description),
                },
            ],
        };
    }
    return null;
};
