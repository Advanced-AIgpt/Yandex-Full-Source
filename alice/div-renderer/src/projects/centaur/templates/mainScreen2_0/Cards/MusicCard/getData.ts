import { IMusicCardProps } from './types';
import { colorDarkGrey } from '../../../../style/constants';
import { ICardDataAdapter } from '../types';

const playlists: readonly [string, string][] = [
    ['Плейлист дня', '#31AD54'],
    ['Мне нравится', '#E53866'],
    ['Тайник', '#186FD3'],
    ['Подкасты', '#036292'],
    ['Дежавю', '#A019D1'],
    ['Премьера', '#D1673E'],
    ['Плейлист с Алисой', '#2D0B87'],
    ['Мой 2021', '#C20146'],
    ['Плейлист семьи', '#FE8853'],
];

export const getMusicCardData: ICardDataAdapter<IMusicCardProps> = function getMusicCardData(card, requestState) {
    if (typeof card.MusicCardData !== 'undefined' && card.MusicCardData !== null) {
        const data = card.MusicCardData;

        const knownPlaylist = playlists.find(el => el[0] === data.Name);

        const color = knownPlaylist ? knownPlaylist[1] : colorDarkGrey;

        return {
            type: 'music',
            name: data.Name,
            modified: data.Modified?.value,
            color,
            cover: data.Cover,
            actions: (card.Action && [
                {
                    log_id: 'music_card_action',
                    url: card.Action,
                },
            ]) || undefined,
            preventDefaultLoader: true,
            requestState,
        };
    }
    return null;
};
