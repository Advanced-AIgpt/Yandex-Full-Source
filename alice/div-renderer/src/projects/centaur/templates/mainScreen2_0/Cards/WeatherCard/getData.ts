import { compact } from 'lodash';
import { IWeatherCardProps } from './types';
import { calcCondition, calcDaypart, getWeatherImage } from '../../../weather/common';
import { EnumLayer } from '../../../../actions/client';
import { ICardDataAdapter } from '../types';

export const getWeatherCardData: ICardDataAdapter<IWeatherCardProps> = function getWeatherCardData(card, requestState) {
    if (typeof card.WeatherCardData !== 'undefined' && card.WeatherCardData !== null) {
        const data = card.WeatherCardData;

        return {
            type: 'weather',
            temperature: data.Temperature,
            image: data.Image,
            comment: data.Comment,
            city: data.City,
            bgImage: data.Sunrise &&
                data.Sunrise &&
                data.UserTime ?
                getWeatherImage(
                    calcDaypart(data.Sunrise, data.Sunrise, data.UserTime),
                    calcCondition(data.Condition?.Cloudness ?? 0, data.Condition?.PrecStrength ?? 0),
                ) : undefined,
            layer: EnumLayer.dialog,
            requestState,
            actions: compact([
                card.Action && {
                    log_id: 'weather_action',
                    url: card.Action,
                },
            ]),
        };
    }
    return null;
};
