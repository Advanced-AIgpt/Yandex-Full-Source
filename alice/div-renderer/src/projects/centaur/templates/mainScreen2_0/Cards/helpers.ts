export function getCardMainScreenId({ rowIndex, colIndex }: { rowIndex?: number; colIndex?: number }) {
    return `main_screen_card_${rowIndex || 0}_${colIndex || 0}`;
}
