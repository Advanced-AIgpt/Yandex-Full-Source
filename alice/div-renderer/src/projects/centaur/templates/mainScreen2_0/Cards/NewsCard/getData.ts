import { compact } from 'lodash';
import { INewsCardProps } from './types';
import { EnumLayer } from '../../../../actions/client';
import { getImageByTopic } from '../../../../helpers/newsHelper';
import { ICardDataAdapter } from '../types';

export const getNewsCardData: ICardDataAdapter<INewsCardProps> = function getNewsCardData(card, requestState) {
    if (typeof card.NewsCardData !== 'undefined' && card.NewsCardData !== null) {
        const data = card.NewsCardData;

        return {
            type: 'news',
            title: data.Title,
            content: data.Content,
            image: data.ImageUrl || getImageByTopic(data.Topic) || undefined,
            layer: EnumLayer.dialog,
            requestState,
            actions: compact([
                card.Action && {
                    log_id: 'news_card_action',
                    url: card.Action,
                },
            ]),
        };
    }
    return null;
};
