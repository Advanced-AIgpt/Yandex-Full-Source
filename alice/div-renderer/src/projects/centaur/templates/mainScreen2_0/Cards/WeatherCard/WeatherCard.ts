import { ContainerBlock, Div, FixedSize, ImageBackground, ImageBlock, TextBlock, WrapContentSize } from 'divcard2';
import { IWeatherCardProps } from './types';
import { text28m, title90m } from '../../../../style/Text/Text';
import { colorWhite, colorWhiteOpacity50 } from '../../../../style/constants';
import { BasicTextCard } from '../BasicTextCard';
import BasicErrorCard from '../BasicErrorCard';
import { logger } from '../../../../../../common/logger';
import { getTemperatureString } from '../../../../helpers/weather/weather';
import { getCardMainScreenId } from '../helpers';

export default function WeatherCard({
    city,
    comment,
    image,
    temperature,
    bgImage,
    actions,
    longtap_actions,
    rowIndex,
    colIndex,
}: Partial<IWeatherCardProps>): Div {
    if (!city || !comment || !image || typeof temperature === 'undefined' || temperature === null) {
        logger.error('The weather card should have a city, a comment, a picture and a temperature, but some of this was not transmitted');

        return BasicErrorCard({
            rowIndex,
            colIndex,
            title: 'Погода',
            description: 'Ошибка сети. Не могу показать прозноз. Проверьте интернет',
            actions,
            longtap_actions,
        });
    }

    const cardData: Parameters<typeof BasicTextCard>[0] = {
        id: getCardMainScreenId({ rowIndex, colIndex }),
        title: city,
        items: [
            new ContainerBlock({
                orientation: 'horizontal',
                content_alignment_vertical: 'center',
                content_alignment_horizontal: 'left',
                items: [
                    new TextBlock({
                        ...title90m,
                        line_height: 90,
                        text_color: colorWhite,
                        width: new WrapContentSize(),
                        text: getTemperatureString(temperature),
                    }),
                    new ImageBlock({
                        height: new FixedSize({ value: 100 }),
                        width: new FixedSize({ value: 100 }),
                        margins: {
                            left: 10,
                        },
                        image_url: image,
                    }),
                ],
            }),
            new TextBlock({
                ...text28m,
                text_color: colorWhiteOpacity50,
                text: comment,
            }),
        ],
        actions,
        longtap_actions,
    };

    if (bgImage) {
        cardData.background = [
            new ImageBackground({
                image_url: bgImage,
                scale: 'fill',
                preload_required: 1,
                content_alignment_horizontal: 'center',
                content_alignment_vertical: 'center',
            }),
        ];
    }

    return BasicTextCard(cardData);
}
