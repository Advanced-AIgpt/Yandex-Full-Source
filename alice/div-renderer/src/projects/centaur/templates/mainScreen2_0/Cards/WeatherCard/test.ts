import { AnonymizeDataForSnapshot } from '../../../../../../common/helpers/dev';
import WeatherCard from './WeatherCard';
import { NAlice } from '../../../../../../protos';
import { getWeatherCardData } from './getData';
import { logger } from '../../../../../../common/logger';
const TCardData = NAlice.NData.TCentaurMainScreenMyScreenData.TColumn.TCardData;

jest.mock('../../../../../../common/logger');

describe('Main screen empty card', () => {
    it('should get actual data', () => {
        expect(getWeatherCardData(new TCardData({}))).toBeNull();

        expect(getWeatherCardData(new TCardData({
            YouTubeCardData: {},
        }))).toBeNull();

        expect(getWeatherCardData(new TCardData({
            YouTubeCardData: {},
            WeatherCardData: {
            },
        }))).not.toBeNull();

        expect(getWeatherCardData(new TCardData({
            WeatherCardData: {
                Temperature: -3,
                City: 'Москва',
                Image: '1',
                Comment: 'Пасмурно.\nОщущается как -5°',
            },
        }))).not.toBeNull();
    });

    it('should match snapshot', () => {
        expect(AnonymizeDataForSnapshot(WeatherCard({
            type: 'weather',
            temperature: -3,
            city: 'Москва',
            image: '1',
            comment: 'Пасмурно.\nОщущается как -5°',
        }))).toMatchSnapshot();

        expect(logger.error).toHaveBeenCalledTimes(0);
    });

    it('should match snapshot with error', () => {
        expect(AnonymizeDataForSnapshot(WeatherCard({
            type: 'weather',
            temperature: undefined,
            city: '',
            image: '',
            comment: 'Пасмурно.\nОщущается как -5°',
        }))).toMatchSnapshot();

        expect(logger.error).toHaveBeenCalled();
    });
});
