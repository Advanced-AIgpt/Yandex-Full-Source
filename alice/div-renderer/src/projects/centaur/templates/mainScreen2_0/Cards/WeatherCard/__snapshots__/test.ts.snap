// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main screen empty card should match snapshot 1`] = `
Object {
  "background": Array [
    Object {
      "color": "#1aeaf6ec",
      "type": "solid",
    },
  ],
  "border": Object {
    "corner_radius": 28,
  },
  "height": Object {
    "type": "match_parent",
    "weight": 1,
  },
  "id": "main_screen_card_0_0",
  "items": Array [
    Object {
      "items": Array [
        Object {
          "font_size": 32,
          "font_weight": "medium",
          "line_height": 40,
          "text": "Москва",
          "text_color": "#80ffffff",
          "type": "text",
        },
      ],
      "orientation": "horizontal",
      "type": "container",
    },
    Object {
      "delimiter_style": Object {
        "color": "#0fff",
      },
      "height": Object {
        "type": "match_parent",
      },
      "type": "separator",
    },
    Object {
      "content_alignment_horizontal": "left",
      "content_alignment_vertical": "center",
      "items": Array [
        Object {
          "font_size": 90,
          "font_weight": "medium",
          "line_height": 90,
          "text": "–3°",
          "text_color": "#fff",
          "type": "text",
          "width": Object {
            "type": "wrap_content",
          },
        },
        Object {
          "height": Object {
            "type": "fixed",
            "value": 100,
          },
          "image_url": "1",
          "margins": Object {
            "left": 10,
          },
          "type": "image",
          "width": Object {
            "type": "fixed",
            "value": 100,
          },
        },
      ],
      "orientation": "horizontal",
      "type": "container",
    },
    Object {
      "font_size": 28,
      "font_weight": "medium",
      "line_height": 36,
      "text": "Пасмурно.
Ощущается как -5°",
      "text_color": "#80ffffff",
      "type": "text",
    },
  ],
  "paddings": Object {
    "bottom": 24,
    "left": 24,
    "right": 24,
    "top": 24,
  },
  "type": "container",
  "width": Object {
    "type": "match_parent",
    "weight": 1,
  },
}
`;

exports[`Main screen empty card should match snapshot with error 1`] = `
Object {
  "background": Array [
    Object {
      "color": "#1aeaf6ec",
      "type": "solid",
    },
  ],
  "border": Object {
    "corner_radius": 28,
  },
  "height": Object {
    "type": "match_parent",
    "weight": 1,
  },
  "id": "main_screen_card_0_0",
  "items": Array [
    Object {
      "items": Array [
        Object {
          "font_size": 32,
          "font_weight": "medium",
          "line_height": 40,
          "text": "Погода",
          "text_color": "#80ffffff",
          "type": "text",
        },
      ],
      "orientation": "horizontal",
      "type": "container",
    },
    Object {
      "delimiter_style": Object {
        "color": "#0fff",
      },
      "height": Object {
        "type": "match_parent",
      },
      "type": "separator",
    },
    Object {
      "font_size": 28,
      "font_weight": "medium",
      "line_height": 36,
      "text": "Ошибка сети. Не могу показать прозноз. Проверьте интернет",
      "text_color": "#80ffffff",
      "type": "text",
    },
  ],
  "paddings": Object {
    "bottom": 24,
    "left": 24,
    "right": 24,
    "top": 24,
  },
  "type": "container",
  "width": Object {
    "type": "match_parent",
    "weight": 1,
  },
}
`;
